Cloudflare/zerotrust_dns/ingest/parser.yml

```
@@ -0,0 +1,154 @@
+    actions:
+      - set:
+          "@timestamp": '{{json_event.message.Datetime}}'
+          destination.ip: '{{json_event.message.DstIP}}'
+          destination.port: '{{json_event.message.DstPort}}'
+          device.id: '{{json_event.message.DeviceID}}'
+          dns.question.name: '{{json_event.message.QueryName}}'
+          dns.question.type: '{{json_event.message.QueryTypeName}}'
+          event.category: ["network"]
+          event.kind: 'event'
+          event.type: ["info"]
+          user.email: '{{json_event.message.Email}}'
+          user.id: '{{json_event.message.UserID}}'
+          source.ip: '{{json_event.message.SrcIP}}'
+          source.port: '{{json_event.message.SrcPort}}'
+          network.protocol: 'dns'
+          network.transport: '{{json_event.message.Protocol}}'
+      
+      - translate:
+        dictionary:
+          1: "A"
+          2: "NS"
+          3: "MD"
+          4: "MF"
+          5: "CNAME"
+          6: "SOA"
+          7: "MB"
+          8: "MG"
+          9: "MR"
+          10: "NULL"
+          11: "WKS"
+          12: "PTR"
+          13: "HINFO"
+          14: "MINFO"
+          15: "MX"
+          16: "TXT"
+          17: "RP"
+          18: "AFSDB"
+          19: "X25"
+          20: "ISDN"
+          21: "RT"
+          22: "NSAP"
+          23: "NSAP-PTR"
+          24: "SIG"
+          25: "KEY"
+          26: "PX"
+          27: "GPOS"
+          28: "AAAA"
+          29: "LOC"
+          30: "NXT"
+          31: "EID"
+          32: "NIMLOC"
+          33: "SRV"
+          34: "ATMA"
+          35: "NAPTR"
+          36: "KX"
+          37: "CERT"
+          38: "A6"
+          39: "DNAME"
+          40: "SINK"
+          41: "OPT"
+          42: "APL"
+          43: "DS"
+          44: "SSHFP"
+          45: "IPSECKEY"
+          46: "RRSIG"
+          47: "NSEC"
+          48: "DNSKEY"
+          49: "DHCID"
+          50: "NSEC3"
+          51: "NSEC3PARAM"
+          52: "TLSA"
+          53: "SMIMEA"
+          55: "HIP"
+          56: "NINFO"
+          57: "RKEY"
+          58: "TALINK"
+          59: "CDS"
+          60: "CDNSKEY"
+          61: "OPENPGPKEY"
+          62: "CSYNC"
+          63: "ZONEMD"
+          64: "SVCB"
+          65: "HTTPS"
+          99: "SPF"
+          100: "UINFO"
+          101: "UID"
+          102: "GID"
+          103: "UNSPEC"
+          104: "NID"
+          105: "L32"
+          106: "L64"
+          107: "LP"
+          108: "EUI48"
+          109: "EUI64"
+          249: "TKEY"
+          250: "TSIG"
+          251: "IXFR"
+          252: "AXFR"
+          253: "MAILB"
+          254: "MAILA"
+          255: "*"
+          256: "URI"
+          257: "CAA"
+          258: "AVC"
+          259: "DOA"
+          260: "AMTRELAY"
+          32768: "TA"
+          32769: "DLV"
+        mapping:
+          json_event.message.QueryType: dns.question.type
+        filter: '{{json_event.message.get("QueryTypeName") == None}}'
```

I think you want to test that `QueryTypeName`
```suggestion
        filter: '{{json_event.message.get("QueryTypeName") != None}}'
```

---

I did meant to test that QueryTypeName did not exist. If it did exist, this step would be useless as dns.question.type would already be set (see line 19)